pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go

```diff
@@ -96,74 +110,94 @@ func (r *HTTPRouteReconciler) gapiToKumaRoute(ctx context.Context, mesh string,
 		selectors = append(selectors, &mesh_proto.Selector{
 			Match: match,
 		})
-	}
 
-	var hostnames []string
+		refRoutes[ref] = &mesh_proto.GatewayRoute{Selectors: selectors}
+	}
 
-	for _, hn := range route.Spec.Hostnames {
-		hostnames = append(hostnames, string(hn))
+	routeConf, routeParentStatus, err := r.gapiToKumaRouteConf(ctx, mesh, route)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	var rules []*mesh_proto.GatewayRoute_HttpRoute_Rule
+	var routes []*mesh_proto.GatewayRoute
+	status := ParentStatuses{}
+
+	for ref, route := range refRoutes {
+		if routeConf != nil {
+			route.Conf = routeConf
+			routes = append(routes, route)
+		}
+
+		status[ref] = gatewayapi.RouteParentStatus{
+			ParentRef:      ref,
+			ControllerName: controllerName,
+			Conditions:     routeParentStatus,
+		}
+	}
 
-	for _, rule := range route.Spec.Rules {
-		var backends []*mesh_proto.GatewayRoute_Backend
+	return routes, status, nil
+}
 
-		for _, backend := range rule.BackendRefs {
-			ref := backend.BackendObjectReference
+func (r *HTTPRouteReconciler) shouldHandleParentRef(
+	ctx context.Context, route kube_client.Object, ref gatewayapi.ParentRef,
+) (bool, error) {
+	namespace := route.GetNamespace()
+	if ns := ref.Namespace; ns != nil {
+		namespace = string(*ns)
+	}
 
-			destination, err := r.gapiToKumaRef(ctx, mesh, namespace, ref)
-			if err != nil {
-				return nil, err
-			}
+	gateway := &gatewayapi.Gateway{}
 
-			backends = append(backends, &mesh_proto.GatewayRoute_Backend{
-				// Weight has a default of 1
-				Weight:      uint32(*backend.Weight),
-				Destination: destination,
-			})
-		}
+	gvks, _, err := r.Scheme.ObjectKinds(gateway)
+	if err != nil {
+		return false, err
+	}
 
-		var matches []*mesh_proto.GatewayRoute_HttpRoute_Match
+	gk := gvks[0].GroupKind()
 
-		for _, match := range rule.Matches {
-			kumaMatch, err := gapiToKumaMatch(match)
-			if err != nil {
-				return nil, errors.Wrap(err, "couldn't convert match")
-			}
+	if *ref.Group != gatewayapi.Group(gk.Group) || *ref.Kind != gatewayapi.Kind(gk.Kind) {
+		return false, nil
+	}
```

Comment:
```suggestion
	if *ref.Group != gatewayapi.Group(gatewayapi.GroupName) || *ref.Kind != gatewayapi.Kind("Gateway") {
		return false, nil
	}
```

Reply:
I have a pretty strong aversion to using constant strings to represent types, as a compromise for readability I'll change it to `reflect.TypeOf.Elem.Name`.